pkg/transport/queuestorage/queuestorage.go

```diff
@@ -14,34 +14,36 @@ import (
 	"github.com/tidwall/sjson"
 )
 
-var (
-	MaxFileSize int64 = 100 * 1024 * 1024 // 100MB
-	MaxRows     int64 = 1_000
-	MaxFileAge        = 1 * time.Hour
-)
-
 type QueueStorage struct {
 	queue   queue.QueueBackend
 	storage filestore.StorageBackend
 
 	DataDir string
 	Workers int
 
-	fws         map[string]*FileWriter
-	fwsMu       sync.Mutex
-	closedFiles chan FileEvent
+	fws          map[string]*FileWriter
+	fwsMu        sync.Mutex
+	closedFiles  chan FileWriterInfo
+	timeProvider func() time.Time
 
 	wg   sync.WaitGroup
 	done chan bool
 }
 
-func NewQueueStorageTransport(queue queue.QueueBackend, storage filestore.StorageBackend) *QueueStorage {
+type QueueStorageTransportParam struct {
+	Queue        queue.QueueBackend
+	Storage      filestore.StorageBackend
+	TimeProvider func() time.Time
```

Comment:
What is the purpose of a `TimeProvider`? To me, it seems like the file lifespan time should just be a configuration for the QueueStorage config, ie, `expiration_seconds` or something.

Reply:
The `TimeProvider() time.Time` is essential for mocking clocks (e.g `time.Now()` et al). By doing so we can set accurately test logic that uses the time functionality.

Reply:
I understand. I generally prefer not to use mocks. Instead of mocking the clock, let's set up test configurations that use a 1s (or less) time interval.